smtp.go

```diff
@@ -32,14 +59,31 @@ func (g *Glutton) HandleSMTP(conn net.Conn) {
 		bufin:  bufio.NewReader(conn),
 		bufout: bufio.NewWriter(conn),
 	}
-	client.w("220 Welcome")
-	g.logger.Infof("[smpt    ] Payload 1: %q", client.r(g))
-	client.w("250 Is it me?")
-	g.logger.Infof("[smpt    ] Payload 2: %q", client.r(g))
-	client.w("250 Sender")
-	g.logger.Infof("[smpt    ] Payload 3: %q", client.r(g))
-	client.w("250 Recipient")
-	g.logger.Infof("[smpt    ] Payload 4: %q", client.r(g))
-	client.w("354 Ok Send data ending with <CRLF>.<CRLF>")
-	g.logger.Infof("[smpt    ] Payload 5: %q", client.r(g))
+	rwait()
+	client.w("220 Welcome!")
+	for {
+		query := strings.Trim(client.r(g), "\r\n")
+		g.logger.Infof("[smtp    ] Payload : %q", query)
+		if strings.HasPrefix(query, "HELO ") {
+			rwait()
+			client.w("250 Hello! Pleased to meet you.")
+		} else if validateMail(query) {
+			rwait()
+			client.w("250 OK")
+		} else if validateRCPT(query) {
+			rwait()
+			client.w("250 OK")
+		} else if strings.Compare(query, "DATA") == 0 {
+			client.w("354 End data with <CRLF>.<CRLF>")
+			for strings.Compare(client.r(g), ".\r\n") != 0 {
```

Comment:
May be you want to add limit over reading data. Like break the loop when condition `len(data) > dataVerbMaxLength` comes true. 
I think `dataVerbMaxLength` should be a constant.

Second Comment: You should log data you are reading